### DISCUSSION TITLE
Getting "json: cannot unmarshal string into Go struct field EmailConn.toaddr of type []string"

### DISCUSSION THREAD

**Discussion by ashx1901** [cybertec-postgresql/pg_timetable#464] [Q&A] [answered]
Hi,

I am new to pg_timetable and I am configuring it for one of our clients. I have created a task with a built-in Send email function but when i am executing it is giving me the below error. Could you please advise what needs to be done to fix it? I am getting this below error in one of my tasks. 

Error: json: cannot unmarshal string into Go struct field EmailConn.toaddr of type []string

Code: 

timetable=> insert into timetable.parameter(task_id, order_id , value)
timetable-> VALUES (8,10,
timetable(> '{
timetable'>      "username": "10.0.15.5",
timetable'>  "serverhost": "mail.abc.com",
timetable'>  "serverport": 25,
timetable'>  "toaddr":"rocky@abc.com",
timetable'>  "subject": "pg_timetable - No Reply"
timetable'>  }'::jsonb);
INSERT 0 1

**Comment by pashagolub** [comment, marked as answer]
Hey,

thanks a lot for your interest. The thing is `toaddr` can hold several recipients. That's why it should be an array, e.g.
```
timetable=> insert into timetable.parameter(task_id, order_id , value)
timetable-> VALUES (8,10,
timetable(> '{
timetable'> "username": "10.0.15.5",
timetable'> "serverhost": "mail.abc.com",
timetable'> "serverport": 25,
timetable'> "toaddr": ["rocky@abc.com", "John Doe <john.doe@email.com>"],
timetable'> "subject": "pg_timetable - No Reply"
timetable'> }'::jsonb);
```

more details: https://pg-timetable.readthedocs.io/en/master/components.html#parameter-value-format

**Reply by ashx1901** [reply]
Thank you so much for your prompt response. I have modified my code as you advised in your above message now I am getting "gomail: could not send email 1: gomail: invalid address "": mail: no address"

**Reply by pashagolub** [reply]
I suppose this is the error from your post server. I'm not sure what exactly is missing here. Maybe you need to specify `"senderaddr":   "user@example.com"`

This doesn't look good to me as well: `"username": "10.0.15.5"`

**Reply by ashx1901** [reply]
I am passing the server name in "senderaddr" here because we need to identify on which server our job has failed. Is there any way we can achieve this?

**Reply by pashagolub** [reply]
Sorry, I'm not sure what you are trying to achieve. And since this is your environment setup issue, I cannot help you so much. You may try to run `pg_timetable` in debug mode (`--log-level=debug`) to see as much additional info as possible.

**Reply by ashx1901** [reply]
It worked now. I was not using "@" in username and in senders address and hence getting the above error. 

timetable=> insert into timetable.parameter(task_id, order_id , value)
timetable-> VALUES (8,10,
timetable(> '{
timetable'> "username": "postgres@10.0.15.5",
timetable'> "serverhost": "mail.abc.com",
timetable'> "serverport": 25,
timetable'> "senderaddr": "postgres@10.0.15.5",
timetable'> "toaddr": ["rocky@abc.com", "John Doe <john.doe@email.com>"],
timetable'> "subject": "pg_timetable - No Reply"
timetable'> }'::jsonb);
